src/main.cpp

```diff
@@ -1,105 +1,106 @@
+using namespace ethernet_driver;
+
+int main() {
+
+  elapsedMillis elapsedTime;
 
-void setup()
-{
   Serial.begin(9600);
   Serial.println("Hello World!");
 
-  connectUdp();
+  // Construct UDP instance
+  EthernetDriver ethernet_driver;
+  ethernet_driver.initDriver();
+  timer.begin(setFlag, 100000);
+
+  while (true) {
+    noInterrupts();
+    if (printFlag) {
+      Serial.println(elapsedTime);
+      elapsedTime = 0;
 
-  encoderMessageTimer.begin([]() { printFlag = true; }, 100000);
+      ethernet_driver.sendMessage();
+
+      printFlag = false;
+    }
+    interrupts();
+
+    delay(5);
+  }
+
+  return 0;
+}
+
+void setFlag() {
+  printFlag = true;
 }
 
-void loop()
-{
-  // create Nanopb messages
+#if defined(ARDUINO_STYLE_FUNCS)
+void setup() {
+
+  Serial.begin(9600);
+  Serial.println("Hello World!");
+
+  connect_udp();
+
+  g_encoder_message_timer.begin(update_flag, 100000);
+}
+
+void loop() {
+#if defined(USING_ENCODER_MESSAGES)
+  /* create Nanopb messages */
   DriveEncodersMessage driveEncodersMessage = DriveEncodersMessage_init_zero;
 
-  // read values from drive motor controllers
+  /* read values from drive motor controllers */
   driveEncodersMessage.frontLeftTicks = 12;
   driveEncodersMessage.frontRightTicks = 15;
   driveEncodersMessage.middleLeftTicks = 22;
   driveEncodersMessage.middleRightTicks = 25;
   driveEncodersMessage.backLeftTicks = 32;
   driveEncodersMessage.backRightTicks = 35;
 
-  // sendEncoderEstimates(driveEncodersMessage);
+  sendEncoderEstimates(driveEncodersMessage);
+#endif
 
-  // send out encoder message
+  /*  Send UDP message
+      Disable interrupts to read / write from g_print_flag
+  */
   noInterrupts();
-  if (printFlag)
-  {
-    Serial.println(elapsedTime);
-    elapsedTime = 0;
+  if (g_print_flag) {
+    Serial.println(g_elapsedTime);
+    g_elapsedTime = 0;
 
-    Udp.beginPacket(ip, localPort);
-    Udp.write("hello");
-    Udp.endPacket();
+    g_udp.beginPacket(g_ip, UDP_PORT);
+    g_udp.write("hello");
+    g_udp.endPacket();
 
-    printFlag = false;
+    g_print_flag = false;
   }
   interrupts();
 
   delay(5);
 }
+#endif
 
-void connectUdp()
-{
-  // start the Ethernet
-  Ethernet.begin(mac, ip);
-
-  // Open serial communications and wait for port to open:
-  Serial.begin(9600);
-  while (!Serial)
-  {
-    delay(10);  // wait for serial port to connect. Needed for native USB port only
-  }
-
-  // Check for Ethernet hardware present
-  if (Ethernet.hardwareStatus() == EthernetNoHardware)
-  {
-    Serial.println("Ethernet shield was not found. Sorry, can't run without hardware. :(");
-    while (true)
-    {
-      delay(10);  // do nothing, no point running without Ethernet hardware
-    }
-  }
-  if (Ethernet.linkStatus() == LinkOFF)
-  {
-    Serial.println("Ethernet cable is not connected.");
-  }
-
-  // start UDP
-  Udp.begin(localPort);
-}
-
-void sendEncoderEstimates(DriveEncodersMessage driveEncodersMessage)
-{
+#if defined(USING_ENCODER_MESSAGES)
```

Comment:
Could you explain why this macro is necessary? I really don't understand the rationale here

Reply:
Deleted